src/theme/DocPage/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import clsx from 'clsx';
+import { HtmlClassNameProvider, ThemeClassNames, PageMetadata } from '@docusaurus/theme-common';
+import {
+  docVersionSearchTag,
+  DocsSidebarProvider,
+  DocsVersionProvider,
+  useDocRouteMetadata,
+} from '@docusaurus/theme-common/internal';
+import DocPageLayout from '@theme/DocPage/Layout';
+import NotFound from '@theme/NotFound';
+import SearchMetadata from '@theme/SearchMetadata';
+import type { Props } from '@theme/DocPage';
+
+function DocPageMetadata(props: Props): JSX.Element {
+  const { versionMetadata } = props;
+  return (
+    <>
+      <SearchMetadata
+        version={versionMetadata.version}
+        tag={docVersionSearchTag(versionMetadata.pluginId, versionMetadata.version)}
+      />
+      <PageMetadata>
+        {versionMetadata.noIndex && <meta name='robots' content='noindex, nofollow' />}
+      </PageMetadata>
+    </>
+  );
+}
+
+export default function DocPage(props: Props): JSX.Element {
+  const { versionMetadata } = props;
+  const currentDocRouteMetadata = useDocRouteMetadata(props);
+  if (!currentDocRouteMetadata) {
+    return <NotFound />;
+  }
+  const { docElement, sidebarName, sidebarItems } = currentDocRouteMetadata;
+  return (
+    <>
+      <DocPageMetadata {...props} />
```

Comment:
Why are we destructuring props here instead of passing it directly?

Reply:
This is swizzled component file from node_modules

Reply:
I guess since this is part of the original component, we can just keep it like this for now.